src/rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -136,35 +136,32 @@ function get_host_list()
 # Prepare env file
 ENV_LIST=env.list
 
-cat <<ENV >> $ENV_LIST
-APP_ID=$APP_ID
-FRAMEWORK_NAME=$FRAMEWORK_NAME
-HADOOP_USER_NAME=$HADOOP_USER_NAME
-PAI_TASK_INDEX=$TASK_INDEX
-PAI_CONTAINER_HOST_IP=$CONTAINER_IP
-PAI_CONTAINER_HOST_PORT_LIST=$CONTAINER_PORTS
-PAI_CONTAINER_ID=$CONTAINER_ID
-ENV
-cat <<'ENV' >> $ENV_LIST
-PAI_DEFAULT_FS_URI={{{ hdfsUri }}}
-PAI_JOB_NAME={{{ jobData.jobName }}}
-PAI_USER_NAME={{{ jobData.userName }}}
-PAI_DATA_DIR={{{ jobData.dataDir }}}
-PAI_OUTPUT_DIR={{{ jobData.outputDir }}}
-PAI_CODE_DIR={{{ jobData.codeDir }}}
-PAI_CURRENT_TASK_ROLE_NAME={{{ taskData.name }}}
-PAI_CURRENT_TASK_ROLE_TASK_COUNT={{{ taskData.taskNumber }}}
-PAI_CURRENT_TASK_ROLE_CPU_COUNT={{{ taskData.cpuNumber }}}
-PAI_CURRENT_TASK_ROLE_MEM_MB={{{ taskData.memoryMB }}}
-PAI_CURRENT_TASK_ROLE_SHM_MB={{{ taskData.shmMB }}}
-PAI_CURRENT_TASK_ROLE_GPU_COUNT={{{ taskData.gpuNumber }}}
-PAI_CURRENT_TASK_ROLE_MIN_FAILED_TASK_COUNT={{{ taskData.minFailedTaskCount }}}
-PAI_CURRENT_TASK_ROLE_MIN_SUCCEEDED_TASK_COUNT={{{ taskData.minSucceededTaskCount }}}
-PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX=$TASK_INDEX
-PAI_JOB_TASK_COUNT={{{ tasksNumber }}}
-PAI_JOB_TASK_ROLE_COUNT={{{ taskRolesNumber }}}
-PAI_JOB_TASK_ROLE_LIST={{{ taskRoleList }}}
-ENV
+echo APP_ID=$APP_ID >> $ENV_LIST
+echo FRAMEWORK_NAME=$FRAMEWORK_NAME >> $ENV_LIST
+echo HADOOP_USER_NAME=$HADOOP_USER_NAME >> $ENV_LIST
+echo PAI_TASK_INDEX=$TASK_INDEX >> $ENV_LIST
+echo PAI_CONTAINER_HOST_IP=$CONTAINER_IP >> $ENV_LIST
+echo PAI_CONTAINER_HOST_PORT_LIST=$CONTAINER_PORTS >> $ENV_LIST
+echo PAI_CONTAINER_ID=$CONTAINER_ID >> $ENV_LIST
+echo PAI_DEFAULT_FS_URI={{ hdfsUri }} >> $ENV_LIST
+echo PAI_JOB_NAME={{ jobData.jobName }} >> $ENV_LIST
+echo PAI_USER_NAME={{ jobData.userName }} >> $ENV_LIST
+echo PAI_DATA_DIR={{ jobData.dataDir }} >> $ENV_LIST
```

Comment:
User can set/modify arbitrary environment variable here because the job data will be unescaped by echo.
(I don't know how to make use of it...)

Reply:
Job env will be only enabled in docker containers, so I think any unpredicted modifications will be safe.